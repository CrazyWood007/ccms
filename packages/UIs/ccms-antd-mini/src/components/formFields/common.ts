
export function formItemLayout (layout: 'horizontal' | 'vertical' | 'inline', fieldType: string, label: string | undefined) {
  const formItemLayout: any = { labelAlign: 'left' }
  if (layout === 'vertical') {
    if (label) {
      formItemLayout.labelCol = { span: 24 }
      formItemLayout.wrapperCol = { span: 24 }
    } else {
      formItemLayout.wrapperCol = { span: 24 }
    }
  } else {
    if (['form', 'group', 'object', 'import_subform', 'tabs'].includes(fieldType)) {
      if (label) {
        formItemLayout.labelCol = { span: 24 }
      }
      formItemLayout.wrapperCol = { span: 24 }
    } else {
      if (label) {
        formItemLayout.labelCol = { span: 6 }
        formItemLayout.wrapperCol = { span: 18 }
      } else {
        formItemLayout.wrapperCol = { span: 24 }
      }
    }
  }

  return formItemLayout
}