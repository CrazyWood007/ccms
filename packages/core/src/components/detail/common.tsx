import React from 'react'
import { ParamConfig } from '../../interface'

import { DetailFieldConfigs as getFieldConfigs } from './'
import ParamHelper from '../../util/param'
/**
 * 详情页表单项基类配置文件格式定义
 * - field:    表单项字段名
 * - label:    表单项名称
 * - defaultValue:  表单项默认值
 * - display: 是否可见
 * - - type:     默认值类型
 * - - * static:   固定值
 * - - * data:     上一步骤数据
 * - - * query:    页面GET方法传参
 * - - * hash:     页面HASH传参
 * - - * interface: 接口入参获取
 * - - value:    默认值（static类型使用）
 * - - field:    字段名（data/query/hash类型使用）（hash类型选填）
 */
export interface DetailFieldConfig {
  field: string
  label: string
  columns?: {
    type: 'span' | 'width'
    value: number | string,
    wrap: boolean
    gutter: number | string
  }
  display?: 'none'
  defaultValue?: ParamConfig,
  condition?: DetailFieldConditionConfig
  layout?: 'horizontal' | 'vertical'
  // styles?: object
}

export interface DetailFieldConditionConfig {
  statement?: string
  params?: Array<{
    field?: string
    data?: ParamConfig
  }>
  debug?: boolean
}

/**
 * 详情页表单项配置文件格式定义 - 枚举
 */
export type DetailFieldConfigs = getFieldConfigs

/**
 * 详情页表单项子类需实现的方法
 * - reset:    表单项重置当前值
 * - set:      表单项设置当前值
 * - get:      表单项获取当前值
 * - validate: 表单项的值校验方法
 */
export interface IDetailField<T> {
  reset: () => Promise<T>
  set: (value: T) => Promise<void>
  get: () => Promise<T>
  validate: (value: T) => Promise<true | DetailFieldError[]>
  fieldFormat: () => Promise<{}>
}

/**
 * 详情页表单项子类需要的入参
 * - ref:
 * - formLayout:
 * - value:
 * - data:
 * - step:
 * - config:
 * - onChange:
 */
export interface DetailFieldProps<C extends DetailFieldConfig, T> {
  // 挂载事件
  ref: (instance: DetailField<C, {}, any> | null) => void
  formLayout: 'horizontal' | 'vertical'
  value: T,
  record: { [field: string]: any },
  data: any[],
  step: number,
  config: C
  // TODO 待删除
  onChange: (value: T) => Promise<void>
  // 事件：设置值
  onValueSet: (path: string, value: T, validation: true | DetailFieldError[]) => Promise<void>
  // // 事件：置空值
  onValueUnset: (path: string, validation: true | DetailFieldError[]) => Promise<void>
  // 事件：修改值 - 列表 - 追加
  onValueListAppend: (path: string, value: any, validation: true | DetailFieldError[]) => Promise<void>
  // 事件：修改值 - 列表 - 删除
  onValueListSplice: (path: string, index: number, count: number, validation: true | DetailFieldError[]) => Promise<void>
  baseRoute: string,
  loadDomain: (domain: string) => Promise<string>
}

/**
 * 详情页配置接口获取数据需要的入参
* - url: 请求地址
* - method: 请求类型
* - withCredentials?: 跨域是否提供凭据信息
* - response: 返回值
* - format?: 格式化返回值
* - responseArrayKey?: format === 'array' 时配置 key 值
* - responseArrayValue?: format === 'array' 时配置 value 值
 */
export interface DetailFieldInterface {
  interface?: {
    url: string
    method: 'GET' | 'POST' | 'get' | 'post'
    withCredentials?: boolean
    response: string
    format?: 'array' | 'key'
    responseArrayKey?: string
    responseArrayValue?: string
  }
}

/**
 * 详情项基类
 * - C: 表单项的配置文件类型
 * - E: 表单项的渲染方法入参
 * - T: 表单项的值类型
 * - S: 表单项的扩展状态
 */
export class DetailField<C extends DetailFieldConfig, E, T, S = {}> extends React.Component<DetailFieldProps<C, T>, S> implements IDetailField<T> {
  static defaultProps = {
    config: {}
  };

  /**
   * 获取默认值
   */
  defaultValue = async () => {
    const {
      config
    } = this.props
    if (config.defaultValue !== undefined) {
      return ParamHelper(config.defaultValue, { record: this.props.record, data: this.props.data, step: this.props.step })
    }

    return undefined
  }

  reset: () => Promise<T> = async () => {
    return this.defaultValue()
  };

  set: (value: T) => Promise<void> = async (value) => {
    const {
      onChange
    } = this.props
    if (onChange) {
      onChange(value)
    }
  };

  get: () => Promise<T> = async () => {
    return this.props.value
  }

  validate: (value: T) => Promise<true | DetailFieldError[]> = async () => {
    return true
  };

  fieldFormat: () => Promise<{}> = async () => {
    return {}
  }

  didMount: () => Promise<void> = async () => { }

  renderComponent = (props: E) => {
    return <React.Fragment>
      当前UI库未实现该展示类型
    </React.Fragment>
  }

  render = () => {
    return (<React.Fragment>
      当前UI库未实现该展示类型
    </React.Fragment>)
  }
}

export class DetailFieldError {
  message: string
  constructor(message: string) {
    this.message = message
  }
}
