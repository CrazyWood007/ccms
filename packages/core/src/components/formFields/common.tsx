import React from 'react'
import { ColumnsConfig, ParamConfig } from '../../interface'

import { FieldConfigs as getFieldConfigs } from './'
import ParamHelper from '../../util/param'
import { updateCommonPrefixItem } from '../../util/value'
import { ConditionConfig } from '../../util/condition'
import { StatementConfig } from '../../util/statement'
import { isEqual, get } from 'lodash'

/**
 * 表单项基类配置文件格式定义
 * - field:    表单项字段名
 * - type:     表单项类型
 * - * text:     短文本类型
 * - * form:     子表单类型
 * - label:    表单项名称
 * - required: 表单项必填
 * - readonly: 表单项只读
 * - disabled: 表单项不可编辑
 * - default:  表单项默认值  // 改为defaultValue
 * - - type:     默认值类型
 * - - * static:   固定值
 * - - * data:     上一步骤数据
 * - - * query:    页面GET方法传参
 * - - * hash:     页面HASH传参
 * - - * interface: 接口入参获取
 * - - value:    默认值（static类型使用）
 * - - field:    字段名（data/query/hash类型使用）（hash类型选填）
 */
export interface FieldConfig {
  field: string
  label: string
  required?: boolean
  readonly?: boolean
  disabled?: boolean
  display?: 'none'
  defaultValue?: ParamConfig,
  condition?: ConditionConfig
  extra?: StatementConfig
  columns?: ColumnsConfig
  styles?: object
}

/**
 * 表单项配置文件格式定义 - 枚举
 */
export type FieldConfigs = getFieldConfigs

/**
 * 表单项子类需实现的方法
 * - reset:    表单项重置当前值
 * - set:      表单项设置当前值
 * - get:      表单项获取当前值
 * - validate: 表单项的值校验方法
 */
export interface IField<T> {
  reset: () => Promise<T>
  set: (value: T) => Promise<void>
  get: () => Promise<T>
  validate: (value: T) => Promise<true | FieldError[]>
  fieldFormat: () => Promise<{}>
}

/**
 * 表单项子类需要的入参
 * - ref:
 * - formLayout:
 * - value:
 * - data:
 * - step:
 * - config:
 * - onChange:
 */
export interface FieldProps<C extends FieldConfig, T> {
  // 挂载事件
  ref: (instance: Field<C, {}, any> | null) => void
  // 挂载引用
  form: React.ReactNode
  formLayout: 'horizontal' | 'vertical' | 'inline'
  value: T,
  record: { [field: string]: any },
  data: any[],
  config: C
  // TODO 待删除
  onChange: (value: T) => Promise<void>
  // 事件：设置值  noPathCombination：为true时不做路径拼接
  onValueSet: (path: string, value: T, validation: true | FieldError[], options?: { noPathCombination?: boolean }) => Promise<void>
  // 事件：置空值
  onValueUnset: (path: string, validation: true | FieldError[], options?: { noPathCombination?: boolean }) => Promise<void>
  // 事件：修改值 - 列表 - 追加
  onValueListAppend: (path: string, value: any, validation: true | FieldError[], options?: { noPathCombination?: boolean }) => Promise<void>
  // 事件：修改值 - 列表 - 删除
  onValueListSplice: (path: string, index: number, count: number, validation: true | FieldError[], options?: { noPathCombination?: boolean }) => Promise<void>
  // 事件：修改值 - 列表 - 修改顺序
  onValueListSort: (path: string, index: number, sortType: 'up' | 'down', validation: true | FieldError[], options?: { noPathCombination?: boolean }) => Promise<void>
  baseRoute: string,
  containerPath: string, // 容器组件所在路径以字段拼接展示  1.3.0新增
  onReportFields?: (field: string) => Promise<void> // 向父组件上报依赖字段  1.3.0新增
  step: { [field: string]: any } // 传递formValue
  loadDomain: (domain: string) => Promise<string>
}

/**
 * 表单项配置接口获取数据需要的入参
* - url: 请求地址
* - method: 请求类型
* - withCredentials?: 跨域是否提供凭据信息
* - response: 返回值
* - format?: 格式化返回值
* - responseArrayKey?: format === 'array' 时配置 key 值
* - responseArrayValue?: format === 'array' 时配置 value 值
 */
export interface FieldInterface {
  interface?: {
    url: string
    method: 'GET' | 'POST' | 'get' | 'post'
    withCredentials?: boolean
    response: string
    format?: 'array' | 'key'
    responseArrayKey?: string
    responseArrayValue?: string
  }
}

/**
 * 表单项基类
 * - C: 表单项的配置文件类型
 * - E: 表单项的渲染方法入参
 * - T: 表单项的值类型
 * - S: 表单项的扩展状态
 */
export class Field<C extends FieldConfig, E, T, S = {}> extends React.Component<FieldProps<C, T>, S> implements IField<T> {
  dependentFields: string[] = [] // 组件param依赖字段存放数组  1.3.0新增
  static defaultProps = {
    config: {}
  };

  /**
   * 获取默认值
   */
  defaultValue : () => Promise<any> = async () => {
    const {
      config
    } = this.props
    if (config.defaultValue !== undefined) {
      return ParamHelper(config.defaultValue, { record: this.props.record, data: this.props.data, step: this.props.step }, this)
    }

    return undefined
  }

  reset: () => Promise<T> = async () => {
    return this.defaultValue()
  };

  set: (value: any) => Promise<any> = async (value) => {
    return value
  };

  get: () => Promise<T> = async () => {
    return this.props.value
  }

  validate: (value: T) => Promise<true | FieldError[]> = async () => {
    return true
  };

  fieldFormat: () => Promise<{}> = async () => {
    return {}
  }

  didMount: () => Promise<void> = async () => { }

  /**
   * 上报param依赖字段名称
   * @param field
   */
  handleReportFields: (field: string) => void = async (field) => {
    const update: string[] | boolean = updateCommonPrefixItem(this.dependentFields, field)
    if (typeof update === 'boolean') return
    this.dependentFields = update
    this.props.onReportFields && await this.props.onReportFields(field)
  }

  renderComponent = (props: E) => {
    return <React.Fragment>
      当前UI库未实现该表单类型
    </React.Fragment>
  }

  shouldComponentUpdate (nextProps: FieldProps<C, T>, nextState: S) {
    // console.log('nextProps', nextProps, this.props, nextProps.value == this.props.value);

    const dependentFieldsArr = this.dependentFields
    // console.log('dependentFieldsArr',dependentFieldsArr);
    let dependentIsChange = false
    if (dependentFieldsArr && dependentFieldsArr.length) {
      for (let i = dependentFieldsArr.length; i >= 0; i--) {
        const nextDependentField = get(nextProps.step, dependentFieldsArr[i])
        const currentDependentField = get(this.props.step, dependentFieldsArr[i])

        if ((nextDependentField || currentDependentField) && nextDependentField !== currentDependentField) {
          dependentIsChange = true
          break
        }
      }
    }

    /**
     * data提交前不变, 去掉这项的比较
     * record也不比较，需要比较的话就在dependentFieldsArr取出record绝对路径
     * */
    if (!dependentIsChange && isEqual(this.state, nextState) && nextProps.value === this.props.value && this.props.config === nextProps.config) {
      // console.log('no update' );
      return false
    }
    return true
  }

  render = () => {
    return (<React.Fragment>
      当前UI库未实现该表单类型
    </React.Fragment>)
  }
}

export interface DisplayProps<C extends FieldConfig, T> {
  value: T,
  record: { [field: string]: any },
  data: any[],
  step: { [field: string]: any }
  config: C,
  // 事件：设置值
  onValueSet: (path: string, value: T, options?: { noPathCombination?: boolean }) => Promise<void>
  // 事件：置空值
  onValueUnset: (path: string, options?: { noPathCombination?: boolean }) => Promise<void>
  // 事件：修改值 - 列表 - 追加
  onValueListAppend: (path: string, value: any, options?: { noPathCombination?: boolean }) => Promise<void>
  // 事件：修改值 - 列表 - 删除
  onValueListSplice: (path: string, index: number, count: number, options?: { noPathCombination?: boolean }) => Promise<void>
  baseRoute: string,
  loadDomain: (domain: string) => Promise<string>
}

export class Display<C extends FieldConfig, E, T, S = {}> extends React.Component<DisplayProps<C, T>, S> {
  defaultValue = async () => {
    const {
      config
    } = this.props
    if (config.defaultValue !== undefined) {
      return ParamHelper(config.defaultValue, { record: this.props.record, data: this.props.data, step: this.props.step })
    }

    return undefined
  }

  reset: () => Promise<T> = async () => {
    return this.defaultValue()
  };

  set: (value: any) => Promise<any> = async (value) => {
    return value
  };

  get: () => Promise<T> = async () => {
    return this.props.value
  }

  renderComponent = (props: E) => {
    return <React.Fragment>
      当前UI库未实现该表单类型
    </React.Fragment>
  }

  didMount: () => Promise<void> = async () => { }

  render = () => {
    return (<React.Fragment>
      当前UI库未实现该表单类型
    </React.Fragment>)
  }
}

export class FieldError {
  message: string
  constructor (message: string) {
    this.message = message
  }
}
