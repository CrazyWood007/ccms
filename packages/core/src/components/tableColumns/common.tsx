import React from 'react'
import { componentType } from '.'

export interface ColumnConfig extends componentType {
  field: string
  label: string,
  defaultValue?: string,
  style?: {
    color?: string
    fontSize?: number
    prefix?: string
    postfix?: string
    customStyle?: object
  }
}

/**
 * 表格项项子类需实现的方法
 * - get:      表格项获取展示值
 */
export interface IColumn<T> {
  getValue: () => Promise<T>
}

export interface ColumnProps<T> {
  ref: (instance: Column<T, any> | null) => void
  record: { [field: string]: any }
  value: any
  data: any[],
  step: number,
  config: T
}

interface ColumnState {
}

export default class Column<T, E> extends React.Component<ColumnProps<T>, ColumnState> implements IColumn<T> {
  constructor (props: ColumnProps<T>) {
    super(props)
    this.state = {}
  }
  
  static defaultProps = {
    config: {}
  }

  getValue= ()=>{
    return this.props.value
  }

  renderComponent = (props: E) => {
    return <React.Fragment>
      您当前使用的UI版本没有实现Column组件。
    </React.Fragment>
  }

  render = () => {
    return (<></>)
  }
}
