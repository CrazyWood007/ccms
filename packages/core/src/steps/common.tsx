import React from 'react'
import { CCMSConfig } from '../main'

/**
 * 页面流转步骤基类配置定义
 * - type: 类型，对应各子类
 */
export interface StepConfig {
}

/**
 * 页面步骤基类 - 入参格式
 * - ref:      页面步骤示例
 * - data:     各步骤数据
 * - step:     当前步骤索引
 * - config:   当前步骤配置文件
 * - onSubmit: 页面步骤的提交事件
 * - onView:   页面步骤的界面切换事件
 */
export interface StepProps<C extends StepConfig> {
  ref: (instance: Step<C> | null) => void
  data: any[]
  step: number
  config: C
  onChange?: (data: any) => Promise<void>
  onSubmit: (data: any, unmountView?: boolean) => Promise<void>
  onMount: () => Promise<void>
  onUnmount: (reload?: boolean, data?: any) => Promise<void>
  checkPageAuth: (pageID: any) => Promise<boolean>
  loadPageURL: (pageID: any) => Promise<string>
  loadPageFrameURL: (pageID: any) => Promise<string>
  loadPageConfig: (pageID: any) => Promise<CCMSConfig>
  baseRoute: string
  loadDomain: (domain: string) => Promise<string>
  handlePageRedirect?: (path: string) => void
}

/**
 * 页面步骤基类
 */
export default class Step<C extends StepConfig, S = {}> extends React.Component<StepProps<C>, S> {
  static defaultProps = {
    config: {
    }
  };

  stepPush = () => {
    this.props.onMount()
  }

  stepPop = (reload: boolean = false, data?: any) => {
    if (reload) {
      this.props.onMount()
    } else {
      this.props.onUnmount()
    }
  }

  render () {
    return <></>
  }
}
