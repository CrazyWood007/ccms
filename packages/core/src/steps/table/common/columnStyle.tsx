import React, { Component } from 'react'

type Props = {
    style: any
    addfix: boolean
};

export default class ColumnStyleComponent extends Component<Props, {}> {
    render() {
        const { style, addfix = true } = this.props
        const reSetStyle = Object.assign({}, { fontSize: style?.fontSize, color: style?.color }, style?.customStyle);
        return addfix ? <div style={reSetStyle}>
           {style?.prefix || ''}{this.props.children}{style?.postfix || ''}
        </div> : <div style={reSetStyle}>
            {this.props.children}
        </div>
    }
}

