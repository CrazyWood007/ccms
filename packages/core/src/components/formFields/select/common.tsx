import { ReactNode } from 'react'
import EnumerationHelper, { EnumerationOptionsConfig, InterfaceEnumerationOptionsKVConfig, InterfaceEnumerationOptionsListConfig } from '../../../util/enumeration'
import InterfaceHelper from '../../../util/interface'
import { Field, FieldConfig, FieldProps, IField, Display, DisplayProps } from '../common'

export interface SelectFieldConfig extends FieldConfig {
  options?: EnumerationOptionsConfig
  defaultSelect?: boolean | number
}

export interface ISelectFieldOption {
  value: string | number | boolean,
  label: ReactNode,
  children?: Array<ISelectFieldOption>
}

interface SelectSingleFieldState {
  options: Array<{
    value: string | number | boolean
    label: string
    [extra: string]: any
  }>
}

export default class SelectField<C extends SelectFieldConfig, E, T> extends Field<C, E, T, SelectSingleFieldState> implements IField<T> {
  interfaceHelper = new InterfaceHelper()

  constructor(props: FieldProps<C, T>) {
    super(props)

    this.state = {
      options: []
    }
  }

  options = (
    config: EnumerationOptionsConfig | undefined
  ) => {
    if (config) {
      EnumerationHelper.options(
        config,
        (config, source) => this.interfaceHelper.request(config, source, { record: this.props.record, data: this.props.data, step: this.props.step }, { loadDomain: this.props.loadDomain }, this),
        { record: this.props.record, data: this.props.data, step: this.props.step }
      ).then((options) => {
        if (JSON.stringify(this.state.options) !== JSON.stringify(options)) {
          this.setState({
            options
          })
        }
      })

      return this.state.options
    }
    return []
  }
}

export class SelectDisplay<C extends SelectFieldConfig, E, T> extends Display<C, E, T, SelectSingleFieldState> {
  interfaceHelper = new InterfaceHelper()

  constructor(props: DisplayProps<C, T>) {
    super(props)

    this.state = {
      options: []
    }
  }

  options = (
    config: EnumerationOptionsConfig | undefined
  ) => {
    if (config) {
      EnumerationHelper.options(
        config,
        (config, source) => this.interfaceHelper.request(config, source, { record: this.props.record, data: this.props.data, step: this.props.step }, { loadDomain: this.props.loadDomain }),
        { record: this.props.record, data: this.props.data, step: this.props.step }
      ).then((options) => {
        if (JSON.stringify(this.state.options) !== JSON.stringify(options)) {
          this.setState({
            options
          })
        }
      })

      return this.state.options
    }
    return []
  }
}
