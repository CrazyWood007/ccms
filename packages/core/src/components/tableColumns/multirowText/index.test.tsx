import React from 'react'
import { render, cleanup } from '@testing-library/react'
import MultirowColumn, { MultirowColumnConfig } from '.'
import { ColumnProps } from '../common'
import renderer from 'react-test-renderer'

// 默认入参
const defaultProps: ColumnProps<MultirowColumnConfig> = {
    ref: async () => { },
    record: {},
    value: '',
    data: [],
    step: 0,
    config: { type: 'multirowText', field: 'test', label: 'test' }
}

const theTest = (message: string, setvalue: any, config: any, successValue: any) => {
    test(message, () => {
        return new Promise((resolve) => {
            render(
                <MultirowColumn
                    {...Object.assign(defaultProps, setvalue)}
                    config={Object.assign(defaultProps.config, config)}
                    ref={async (ref: any) => {
                        if (ref) {
                            const value = await ref?.getValue('test')
                            expect(value).toEqual(successValue)
                            resolve(true)
                            cleanup()
                        }
                    }}
                />
            )
        })
    })
}

theTest('多行文本- 未配置', { value: "" }, {}, [])
theTest('多行文本- 暂无信息', { value: null }, { defaultValue: '暂无信息', }, ['暂无信息'])

test("多行文本- 无内容渲染 验证dom长度", () => {
    return new Promise((resolve) => {

        const renderTest = renderer.create(<MultirowColumn
            {...Object.assign(defaultProps, { value: null })}
            config={Object.assign(defaultProps.config,{ defaultValue: '暂无信息'})}
            ref={async (ref: any) => {
                if (ref) {
                    const value = await ref?.getValue()
                    expect(value).toEqual(['暂无信息'])
                    cleanup()
                    resolve(true)
                }
            }}
        />)
        const o = renderTest.toJSON()
        const testInstance = renderTest.root;
        expect(testInstance.children.length).toEqual(1)
    })
})

theTest('多行文本- 配置默认格式 单行 非字符串', { value: 1 }, {}, [1])
theTest('多行文本- 配置默认格式 单行 字符串', { value: '1' }, {}, ['1'])
theTest('多行文本- 配置默认格式', { value: '第一行,第二行' }, {}, ["第一行", "第二行"])
theTest('多行文本- 数组', { value: ["第一行", "第二行"] }, {}, ["第一行", "第二行"])
theTest('多行文本- 添加前缀', { value: '第一行,第二行' }, { style: { prefix: "LINE:" } }, ["LINE:第一行", "LINE:第二行"])
theTest('多行文本- 添加后缀', { value: '第一行,第二行' }, { style: { postfix: "end" } }, ["第一行end", "第二行end"])
theTest('多行文本- 添加前后缀', { value: '第一行,第二行' }, { style: { prefix: "LINE:", postfix: "end" } }, ["LINE:第一行end", "LINE:第二行end"])


test("多行文本- 有内容 验证dom长度", () => {
    return new Promise((resolve) => {
        const renderTest = renderer.create(<MultirowColumn
            {...Object.assign(defaultProps, { value: '第一行,第二行' })}
            config={Object.assign(defaultProps.config, { style: { prefix: "", postfix: '' } })}
            ref={async (ref: any) => {
                if (ref) {
                    const value = await ref?.getValue()
                    expect(value).toEqual(["第一行", "第二行"])
                    cleanup()
                    resolve(true)
                }
            }}
        />)
        const o = renderTest.toJSON()
        expect(o).toMatchSnapshot()
        const testInstance = renderTest.root;
        expect(testInstance.children.length).toEqual(2)
    })

})