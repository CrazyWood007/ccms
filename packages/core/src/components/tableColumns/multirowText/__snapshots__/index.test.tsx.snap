// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`多行文本- 有内容 验证dom长度 1`] = `
Array [
  <div>
    您当前使用的UI版本没有实现MultirowColumn组件。
  </div>,
  <div>
    您当前使用的UI版本没有实现MultirowColumn组件。
  </div>,
]
`;
